design-documents/graph-ql/coverage/search.md

```diff
@@ -0,0 +1,79 @@
+# Queries
+
+```graphql
+type Query {
+    multiSearch(phrase: String!, productSize: Int = 10): MultiSearchResponse!
+    #Filter supports multiple clauses which will be wrapped in logical AND operator
+    productSearch(phrase: String!, filter: [SearchClauseInput], sort: [ProductSearchSortInput]): ProductSearchResponse!
+}
+
+input ProductSearchSortInput
+{
+    attribute: String!
+    direction: SortEnum!
+}
+
+# If from or to fields are omitted, $gte or $lte filter will be applied
+input SearchRangeInput {
+    from: Int
+    to: Int
+}
+
+input SearchClauseInput {
+    attribute: String!
+    in: [String]
+    eq: String
+    range: SearchRangeInput
+}
+
+type ProductSearchResponse {
+    items: [ProductSearchItem]
+    hits: Int!
+    facets: [Aggregation]
+    facetsValues: [Aggregation]
+}
+
+type MultiSearchResponse {
+    products: ProductSearchResponse
+    suggestions: [String]
+}
+
+type ProductSearchItem {
+    product: ProductInterface!
+    highlights: [Highlight]
+}
+
+type Highlight {
+    attribute: String!
+    value: String!
+    matchedWords: [String]!
+}
+
+interface Bucket {
+    #Human readable bucket title
+    title: String!
+}
+
+type StatsBucket implements Bucket {
+    #Could be used for filtering and may contain non-human readable data
+    id: ID!
+    min: Int!
+    max: Int!
+}
+
+type ScalarBucket implements Bucket {
+    id: ID!
+    count: Int!
+}
+
+type RangeBucket implements Bucket {
+    from: Float!
+    to: Float!
+    count: Int!
+}
+
+interface Aggregation {
```

Comment:
I don't understand the relation between `SearchClauseInput.attribute` and Aggregation
Can you describe how to achieve scenario:
- search by "Iphpne 11" -> result with color attribute facets
- drill down by color facets

Reply:
`SearchClauseInput` is a filter, you may filter by any attributes and values which were retrieved from aggregations. Thus the main flow will be:
- Perform a search request with some phrase
- Extract aggregations from the response and render them in layered navigation
- If user clicks on some filter in layer navigation section, apply this filter in new query + apply the old search phrase if needed

Reply:
Actually question is how to map Aggregation.? (what the field I need to use) into SearchClauseInput.attribute?

Reply:
They share the same attribute_code